dlib/dnn/loss.h

```diff
@@ -1737,7 +1755,45 @@ namespace dlib
             label_iterator iter
         )
         {
-            loss_multiclass_log_per_pixel_::to_label(input_tensor, sub, iter);
+            DLIB_CASSERT(sub.sample_expansion_factor() == 1);
+
+            const tensor& output_tensor = sub.get_output();
+
+            DLIB_CASSERT(output_tensor.k() >= 1); // Note that output_tensor.k() should match the number of labels.
+            DLIB_CASSERT(output_tensor.k() < std::numeric_limits<uint16_t>::max());
+            DLIB_CASSERT(input_tensor.num_samples() == output_tensor.num_samples());
+
+            const float* const out_data = output_tensor.host();
```

Comment:
In fact I would want to softmax the output tensor, before collecting the scores below. Is introducing a temporary tensor here too wasteful? Is there a better way?